src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,93 @@
+package duke.task;
+
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Abstract superclass of a task that a user has.
+ */
+public abstract class Task {
+    protected static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("dd MMM yyyy");
+    private String description;
+    private boolean isDone;
+
+    /**
+     * Constructor for {@code Task}.
+     *
+     * @param description Description of the {@code Task}.
+     * @param isDone Completion status of the {@code Task}.
+     */
+    protected Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Constructor for {@code Task}. Assumes completion status is {@code false}.
+     *
+     * @param description Description of the {@code Task}.
+     */
+    protected Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Sets the {@code isDone} status variable to {@code true}.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Sets the {@code isDone} status variable to {@code false}.
+     */
+    public void unmarkAsDone() {
+        this.isDone = false;
+    }
+
+    /**
+     * Checks if the {@code Task} description contains the keyword provided.
+     *
+     * @param keyword
+     * @return {@code true} if keyword is present in description, {@code false} otherwise.
+     */
+    public boolean contains(String keyword) {
+        return this.description.contains(keyword);
+    }
+
+    /**
+     * Returns a String representation of the {@code Task} in storage format.
+     */
+    public String getStorageFormat() {
+        return this.getStatusBit() + " | " + this.description;
```

Comment:
Hi Wei Quan, the method name `getStorageFormat` effectively communicates that this method is meant to encode a `Task` as a `String` for storage. Other authors I reviewed have used names such as `stringify`, which is problematic because it does not self-explain the difference between `stringify` and `toString`.

However, I think it would be clearer to use `String.format()` here. It might be unnecessary to use `getStatusBit()`.

```suggestion
        return String.format("%d | %s", this.isDone ? 1 : 0, this.description);
```